encoders/firebase-encoders-reflective/src/main/java/com/google/firebase/encoders/reflective/InternalAnnotationContext.java

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.encoders.reflective;
+
+import com.google.firebase.encoders.EncodingException;
+import com.google.firebase.encoders.annotations.Encodable;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+final class InternalAnnotationContext {
+  private boolean inline;
+  private String decodingKey;
+  private boolean ignored;
+
+  InternalAnnotationContext(AccessibleObject accObj) {
+    if (accObj instanceof Field) {
+      decodingKey = ((Field) accObj).getName();
+    } else if (accObj instanceof Method) {
+      decodingKey = fieldName((Method) accObj);
+    } else {
+      throw new EncodingException("Constructor shouldn't be used to get its decoding key");
+    }
+    this.inline = false;
+    this.ignored = false;
+
+    // internal annotations
+    if (accObj.isAnnotationPresent(Encodable.Ignore.class)) {
+      this.setIgnored(true);
+      return;
+    }
+    if (accObj.isAnnotationPresent(Encodable.Field.class)) {
+      Encodable.Field annotation = accObj.getAnnotation(Encodable.Field.class);
+      if (annotation != null && annotation.name().length() > 0) {
+        this.setDecodingKey(annotation.name());
+      }
+      if (annotation != null) {
+        this.setInline(annotation.inline());
+      }
+    }
+
+    // alias annotations
```

Comment:
looks like this only supports `Ignore` and `Field`, can you add a `TODO` to add support for custom `ExtraProperty` annotations?